lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -339,12 +345,15 @@ def init_api_client
         key = Google::APIClient::PKCS12.load_key(@private_key_path,
                                                  @private_key_passphrase)
         jwt_asserter = Google::APIClient::JWTAsserter.new(
-          @private_key_email, 'https://www.googleapis.com/auth/logging.write',
-          key)
+          @private_key_email, LOGGING_SCOPE, key)
         @client.authorization = jwt_asserter.to_authorization
         @client.authorization.expiry = 3600  # 3600s is the max allowed value
       else
-        @client.authorization = Google::APIClient::ComputeServiceAccount.new
+        if !@application_credentials_path.nil?
+          ENV['GOOGLE_APPLICATION_CREDENTIALS'] = @application_credentials_path
+        end
+        @client.authorization = Google::Auth.get_application_default(
+            LOGGING_SCOPE)
```

Comment:
I think I will do this (get rid of the option), actually.  Better to keep it simple and add it if we need it, since once it is in the wild we'd have to deprecate it to remove it.